[@types/jquery] Fix for issue #52052 - add undefined to get() function

princefishthrower opened the pull request:
As the title states, this should solve the issue brought up here: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/52052

According to the jQuery docs for `get()`:

```
If the value of index is out of bounds — less than the negative number of elements or equal to or greater than the number of elements — it returns undefined.
```

So this typing change brings in that `undefined` case.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://api.jquery.com/get/#get-index
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

johnnyreilly reviewed (approved):
Thanks!

peterblazejewicz commented:
@princefishthrower please make this change in `index.d.ts` (version 3.*), that's not useful to ship this for v2 (you could revert v2 changes altogether IMO).
Also please add a test, a change is OK oterwise, so will be approved promptly I understand.

princefishthrower commented:
@peterblazejewicz - I believe I've completed the requested changes in my latest commit. I reverted v2 changes and added the typing to the newest version, and adjusted tests as needed.

peterblazejewicz commented:
@princefishthrower that would be ok to quickly fix those tests by using non-null assertion operator:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator
(that would present ntent properly IMO), so:
```ts
select = $select.select2().get(0)!;
```
